staging/src/k8s.io/apiserver/pkg/admission/interfaces.go

```diff
@@ -49,6 +49,12 @@ type Attributes interface {
 	GetKind() schema.GroupVersionKind
 	// GetUserInfo is information about the requesting user
 	GetUserInfo() user.Info
+
+	// AddAnnotation sets annotation according to key-value pair. The key should be qualified, e.g., podsecuritypolicy.admission.k8s.io/admit-policy, where
+	// "podsecuritypolicy" is the name of the plugin, "admission.k8s.io" is the name of the organization, "admit-policy" is the key name.
+	// An error is returned if the format of key is invalid.
+	// Both ValidationInterface and MutationInterface are allowed to add Annotations.
```

Comment:
what happens on conflict?

Reply:
If a annotation key is set more than once, it will be overwritten.

One key should always be set only be one admission controller, so conflict seldom happens.

The use case of overwritten behavior:
1. an admission made a decision and set a key.
2. later, it made a new decision and overwrite the key.

Reply:
Added the behavior to comments.